recipe/meta.yaml

```diff
@@ -1,311 +1,309 @@
+        {{ opencv_libs.append("rgbd") or "" }}
+        {{ opencv_libs.append("saliency") or "" }}
+        {{ opencv_libs.append("shape") or "" }}
+        {{ opencv_libs.append("stereo") or "" }}
+        {{ opencv_libs.append("stitching") or "" }}
+        {{ opencv_libs.append("structured_light") or "" }}
+        {{ opencv_libs.append("superres") or "" }}
+        {{ opencv_libs.append("surface_matching") or "" }}
+        {{ opencv_libs.append("text") or "" }}
+        {{ opencv_libs.append("tracking") or "" }}
+        {{ opencv_libs.append("video") or "" }}
+        {{ opencv_libs.append("videoio") or "" }}
+        {{ opencv_libs.append("videostab") or "" }}
+        {{ opencv_libs.append("wechat_qrcode") or "" }}
+        {{ opencv_libs.append("xfeatures2d") or "" }}
+        {{ opencv_libs.append("ximgproc") or "" }}
+        {{ opencv_libs.append("xobjdetect") or "" }}
+        {{ opencv_libs.append("xphoto") or "" }}
+        {{ opencv_libs.append("freetype") or "" }}
+        - export MACOSX_DEPLOYMENT_TARGET={{ MACOSX_DEPLOYMENT_TARGET }}      # [osx]
+        - export CONDA_BUILD_SYSROOT={{ CONDA_BUILD_SYSROOT }}                # [osx]
+        - OPENCV_FLAGS=`pkg-config --cflags opencv4`                          # [unix]
+        - $CXX -std=c++11 $RECIPE_DIR/test.cpp ${OPENCV_FLAGS} -o test        # [unix]
+        - if [[ $(./test) != $PKG_VERSION ]]; then exit 1 ; fi                # [unix]
         {% for each_opencv_lib in opencv_libs %}
-        - test -f ${CONDA_PREFIX}/lib/libopencv_{{ each_opencv_lib }}.dylib                        # [osx]
-        - test -f ${CONDA_PREFIX}/lib/libopencv_{{ each_opencv_lib }}.so                           # [linux]
-        - if not exist %CONDA_PREFIX%\\Library\\bin\\opencv_{{ each_opencv_lib }}{{ version|replace(".","") }}.dll exit 1 # [win]
+        - echo Testing for presence of {{ each_opencv_lib }}
+        - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}${SHLIB_EXT}     # [unix]
+        - if not exist %PREFIX%\\Library\\bin\\opencv_{{ each_opencv_lib }}{{ win_ver_lib }}.dll exit 1  # [win]
+        - echo Found                   {{ each_opencv_lib }}
         {% endfor %}
+        - test -f $PREFIX/lib/libopencv_bioinspired${SHLIB_EXT}  # [unix]
+        - test -f $PREFIX/lib/libopencv_hdf${SHLIB_EXT}          # [unix]
+        - mkdir -p cmake_build_test && pushd cmake_build_test
+        - cmake -G "Ninja" ..
+        - cmake --build . --config Release
+        - popd
 
-  {% if build_variant == "normal" %}
-  # This recipe has merged the outputs for the compiled libraries with the
-  # python bindings that used to be vendored as libopencv and py-opencv. This
-  # was done to try to improve maintainibility. The following two output
-  # sections provide no files other than those already in the opencv package.
-  # These "virtual" packages provide backwards compatibility with recipes
-  # depending on either of the two old packages.
+outputs:
   - name: libopencv
```

Comment:
two `- name` entries right after one another - is this right?

Reply:
From my understanding, the namespace is shared between top-level and libopencv and effectively what is installed from build.sh is used.

Reply:
Yeah, it's something like Charles says

Reply:
ah, I see, I didn't notice that the name is shared there